### PR TITLE
feat: harden release tooling and PTY smoke infra

### PR DESCRIPTION
**Pull request by flyingrobots**
## Summary
- harden PTY smoke tooling against bytecode noise and add a resize/exit lifecycle regression
- add a local PR review status helper for unresolved threads and check summaries
- add a workflow-dispatch release dry run that exercises build, smoke, publish dry-runs, and release notes preview

## Verification
- npm run build
- npm run typecheck:test
- npm test
- npm run pr:review-status -- 41
- npm run smoke:canaries -- --skip-build
- npm run smoke:examples:all

## Summary by CodeRabbit

* **New Features**
  * Added an automated release dry-run workflow with package dry-run publishing and release-notes preview artifact
  * Added a PR review status CLI tool and an npm script for quick PR health checks

* **Tests**
  * Added unit tests for PR review status logic
  * Expanded PTY-driver tests to cover a resize-after-exit edge case

* **Chores**
  * Disabled Python bytecode generation and updated ignore rules

* **Documentation**
  * Updated changelog with PR status and dry-run notes

### CODE REVIEW

File: scripts/pr-review-status.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+export function computeExitCode(summary: CheckSummary, unresolvedCount: number): number {
+  if (summary.failing.length > 0 || unresolvedCount > 0) {
+    return 1;
+  }
+
+  if (summary.pending.length > 0) {
+    return 8;
+  }
+
+  return 0;
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply scripts/pr-review-status.ts:113]: Fixed in commit 006b758. `computeExitCode()` now treats canceled checks the same as failing checks, and the helper prints a dedicated `Canceled checks:` section so canceled required jobs cannot be misreported as green.

File: .github/workflows/release-dry-run.yml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          set -euo pipefail
+          VERSION=$(node -p "require('./packages/bijou/package.json').version")
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+          echo "notes_tag=dry-run-v${VERSION}-${GITHUB_RUN_ID}" >> "$GITHUB_OUTPUT"
+
+      - name: Build
+        run: npm run build
+
+      - name: Type check
+        run: npm run lint
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply .github/workflows/release-dry-run.yml]: Fixed in commit 006b758. The release dry-run workflow step now uses the correct `Lint` label for `npm run lint`, so failure output matches the action being performed.

File: scripts/pr-review-status.ts [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+  readonly link?: string;
+  readonly workflow?: string;
+}
+
+interface ReviewThreadComment {
+  readonly author: { readonly login: string };
+  readonly body: string;
+  readonly path?: string | null;
+  readonly url: string;
+}
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply scripts/pr-review-status.ts:30]: Fixed in commit 006b758. `ReviewThreadComment.author` is now nullable, unresolved finding extraction uses `comment.author?.login ?? "(unknown)"`, and the unit tests cover anonymous review comments so the helper no longer crashes on null authors.

File: .github/workflows/release-dry-run.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                if (!p[depType]) continue;
+                for (const [name, ver] of Object.entries(p[depType])) {
+                  if (name.startsWith('@flyingrobots/bijou') && ver !== '$VERSION') {
+                    console.log('::error::' + p.name + ' has ' + name + '@' + ver + ' in ' + depType + ', expected $VERSION');
+                    process.exitCode = 1;
+                  }
+                }
+              }
+            " || ERRORS=$((ERRORS+1))
+          done
```
[the remark this answers is not in the document]
Comment (flyingrobots) [reply .github/workflows/release-dry-run.yml:59]: Fixed in commit a99bd02. The dry-run metadata validation now checks internal peerDependencies alongside dependencies and devDependencies, so it matches the lock-step version policy enforced by publish.yml.
